### PR TITLE
feat: [SIW-1043] Add `CBOR` decoding assertion function on iOS side

### PR DESCRIPTION
**Pull request by hevelius**
### Short description
This PR adds SwiftCBOR package to easily decoding CBOR assertion on iOS. So this PR introduces a new native function `decodeAssertion`. The output should be send to the backend.

Using CBOR on native side remove from the backend written in ts the computational cost of managed CBOR encoded data.

#### List of Changes

- Add SwiftCBOR
- Updated exmaple app
- Update backend

#### How Has This Been Tested?

- Set value in .env file under example app and backend
- Start metro with `cd example && yarn start`
- Start backend with `cd backend && yarn start`
- Using a supported iOS device open Xcode and start the example app
- Try sequentially all the CTA
- Last CTA `Verify Assertion` should return a value like this `{result: {signCount: 1}}`

### CODE REVIEW

File: ios/IoReactNativeIntegrity.swift [hunk 1]
```diff
@@ -1,11 +1,62 @@
...
+      
+      // Ensure the decoded value is a map
+      guard case let .map(map) = decodedValue else {
+        throw CBORDecodingError.invalidFormat("Expected CBOR map")
+      }
+      
+      // Extract the signature and authenticatorData from the map
+      guard case let .byteString(signatureBytes)? = map["signature"],
+            case let .byteString(authenticatorDataBytes)? = map["authenticatorData"] else {
+        throw CBORDecodingError.invalidFormat("Expected signature and authenticatorData in the CBOR map")
```
Comment (LazyAfternoons) [inline ios/IoReactNativeIntegrity.swift:47]: Given that we are writing this library directly into the bridge code, should we remove this `CBORDecodingError` and just reject the promise with our `ME` object and the specified message? We could then get rid of the custom error.

Comment (hevelius) [reply ios/IoReactNativeIntegrity.swift:47]: Thinking about `CBORDecodingError`, in effect the error will then be captured by the `do..catch` statement. If I can honestly say I wouldn't mind leaving it like that and reject directly in the catch.

Comment (LazyAfternoons) [reply ios/IoReactNativeIntegrity.swift:47]: I see it as a complication we don't need but it's fine, we can leave it like this.

**Review by LazyAfternoons — APPROVED** [review]
LGTM

**Pull request merged**